mbed_lstools/lstools_base.py

```diff
@@ -73,6 +73,8 @@ def __init__(self):
         self.lock_file = os.path.join(MbedLsToolsBase.HOME_DIR,
                                       MbedLsToolsBase.MBEDLS_HOME_DIR, MbedLsToolsBase.MBEDLS_GLOBAL_LOCK)
         self.mbedls_get_mocks()
+        # Make sure retargeting is applied if retarget file exists
+        self.retarget(True)
```

Comment:
Hi @stevew817 I did not propose to pass a flag in the ```retarget()```. What I meant was to move the check ```if opts.skip_retarget:``` inside ```retarget()```. So that we don't have to check it every time we invoke ```retarget()```. But this will involve some more changes to make ```opts``` available inside ```retarget()```. May be have a accessor function to make it available globally. 
This is given that we have ```if opts.skip_retarget:```. We can't change a functionality and leave some options useless.
It would be worth to think if we require option ```opts.skip_retarget```. If we don't then we should also do the cleanup!